lib/grp.gi

```diff
@@ -701,6 +701,18 @@ InstallMethod( AbelianInvariants,
     return inv;
     end );
 
+InstallMethod( AbelianRank ,"generic method for groups", [ IsGroup ],0,
+function(G)
+local a,r;
+  a:=AbelianInvariants(G);
+  r:=Number(a,IsZero);
+  a:=Filtered(a,x->not IsZero(x));
+  if Length(a)=0 then return r; fi;
+  a:=List(Set(Factors(Product(a))),p->Number(a,x->x mod p=0));
```

Comment:
Multiplying prime powers together only to factor them again is wasteful. This could be avoided e.g. as follows:
```
a:=List(Set(a, SmallestRootInt), p->Number(a,x->x mod p=0));
```

Reply:
If this is considered nicer, I'll change it.
In practice this is a purely academic argument as we are talking about factorization into known primes, i.e. trial division.

Reply:
Any review remark I make of course just constitutes my personal opinion, and you are of course free to ignore it.

Anyway, all in all, over the past months I got a strong feeling that my remarks on your PRs are perceived more as nuisance than useful, and as "wasting your time". Therefore, in order to not waste *both* our time, I will refrain from reviewing your PRs for the time being, and leave it to others to take care of that.